src/reflect/scala/reflect/internal/tpe/TypeComparers.scala

```diff
@@ -351,6 +351,7 @@ trait TypeComparers {
   // @assume tp1.isHigherKinded || tp2.isHigherKinded
   def isHKSubType(tp1: Type, tp2: Type, depth: Depth): Boolean = {
     def isSub(ntp1: Type, ntp2: Type) = (ntp1.withoutAnnotations, ntp2.withoutAnnotations) match {
+      case (TypeRef(_, AnyClass, _), PolyType(_, _))                        => true                     // Any is kind-polymorphic
```

Comment:
There you go @sellout, just clone the HK type params and mutate the new symbols. Welcome to scalac!